### PR TITLE
Make option to set vertical tab position

### PR DESCRIPTION
**Pull request by sangwoo108**
Make it possible to move vertical tabs on the right side of browser window.

Resolves https://github.com/brave/brave-browser/issues/30812

### settings page
image

### On right
image

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: app/brave_settings_strings.grdp [hunk 1]
```diff
@@ -83,6 +83,15 @@
...
   </message>
+  <message name="IDS_SETTINGS_APPEARANCE_SETTINGS_BRAVE_TAB_VERTICAL_TAB_POSITION" desc="Title for radio group to set vertical tab position">
+    Position
+  </message>
+  <message name="IDS_SETTINGS_APPEARANCE_SETTINGS_BRAVE_TAB_VERTICAL_TAB_ON_RIGHT" desc="The label for radio button to show vertical tab on right">
+    Show on right
+  </message>
+  <message name="IDS_SETTINGS_APPEARANCE_SETTINGS_BRAVE_TAB_VERTICAL_TAB_ON_LEFT" desc="The label for radio button to show vertical tab on left">
+    Show on left
+  </message>
```
[the remark this answers is not in the document]
Comment (sangwoo108) [reply app/brave_settings_strings.grdp:94]: I agree. Do you want me to change strings for sidebar together?

image

File: app/brave_settings_strings.grdp [hunk 2]
```diff
@@ -83,6 +83,15 @@
   <message name="IDS_SETTINGS_APPEARANCE_SETTINGS_BRAVE_TAB_HOVER_MODE" desc="The label for the settings select controlling what happens when a user hovers over a tab.">
     Tab hover mode
   </message>
+  <message name="IDS_SETTINGS_APPEARANCE_SETTINGS_BRAVE_TAB_VERTICAL_TAB_POSITION" desc="Title for radio group to set vertical tab position">
+    Position
```
Comment (simonhong) [inline app/brave_settings_strings.grdp:87]: Position header seems missing in webui?

File: browser/ui/views/frame/brave_browser_view_layout.cc [hunk 3]
```diff
@@ -249,7 +253,11 @@ void BraveBrowserViewLayout::UpdateContentsContainerInsets(
   // vertical tab itself has sufficient padding.
```
Comment (simonhong) [inline browser/ui/views/frame/brave_browser_view_layout.cc:253]: nit: `Don't need contents container's left or right margin with vertical tab as`

File: browser/ui/views/frame/vertical_tab_strip_widget_delegate_view.cc [hunk 4]
```diff
@@ -106,13 +106,12 @@ void VerticalTabStripWidgetDelegateView::ChildPreferredSizeChanged(
...
   // the web view.
-  if (auto new_host_size = region_view_->GetMinimumSize();
-      new_host_size != host_->GetPreferredSize()) {
-    host_->SetPreferredSize(new_host_size);
-    return;
-  }
+  host_->SetPreferredSize(region_view_->GetMinimumSize());
 
-  // Layout widget manually because host won't trigger layout.
+  // The might could be changed, so we should lay out again.
```
Comment (simonhong) [inline browser/ui/views/frame/vertical_tab_strip_widget_delegate_view.cc:111]:
```suggestion
  // The height could be changed, so we should lay out again.
```

**Review by simonhong — APPROVED** [review]
++ 👍🏼

File: app/brave_settings_strings.grdp (hunk 1 above)
[the remark this answers is not in the document]
Comment (sangwoo108) [reply app/brave_settings_strings.grdp]: Hum, I tried to update it from `settings_strings_override.grdp`, but it's abandoned after running `npm run chromium_rebase_l10n` on both CI and my local machine. @mkarolin could you tell me what's proper way?

[the remark this answers is not in the document]
Comment (mkarolin) [reply app/brave_settings_strings.grdp]: Yes, anything with `_override.grd(p)` is auto-generated and cannot be used for modifications. You would have to add those strings to `brave_settings_strings.grdp` and then in the `chromium_src/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc` override redefine the original IDS with the new brave IDS (you can see examples of that in the override file).

[the remark this answers is not in the document]
Comment (sangwoo108) [reply app/brave_settings_strings.grdp]: Aha! Thanks, it works perfect!

**Review by mkarolin — APPROVED** [review]
`strings` and `chromium_src` ++

**Pull request merged**